lpvartypes.pas

```diff
@@ -1793,6 +1815,13 @@ function TLapeType.Eval(Op: EOperator; var Dest: TResVar; Left, Right: TResVar;
   begin
     Dest := NullResVar;
     Exit(Left);
+  end 
+  else if (op in CompoundOperators) and (lcoCOperators in Compiler.FOptions) then 
+  begin
+    if (not Left.Writeable) then
+      LapeException(lpeCannotAssign, _DocPos);
```

Comment:
This check will be done during the recursive `op_Assign` evaluation. I think you can use a similar one-liner as used above.